[18789] restructure workpackage summary

HDinger opened the pull request:
work package summary table page has a new structure

https://community.openproject.org/work_packages/18789

HDinger committed:
work package summary table page has a new structure

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://hdinger-feature-18789-restructure-workpackage-summary.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://hdinger-feature-18789-restructure-workpackage-summary.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

HDinger commented:
This PR is based on the PR https://github.com/opf/openproject/pull/3343 (apply new table styling). Thus the previous PR should be merged before this one.

Further one problem still exists for all tables: If the table width is too high for the window a scrollbar will be shown. When using the scrollbar, it appears that the overleaping part of the table is not shown correctly. 

![table_overflow](https://cloud.githubusercontent.com/assets/7457313/9385482/04d64c12-4757-11e5-8651-0fd7384d6a46.png)

HDinger committed:
fixed broken specs

HDinger committed:
avoids the punctuation of icons

ulferts committed:
specify interactive-table to be used on table tag only

It was formerly used on the container elements as well which looked to be ok on first glance but failed to calculate the
table width correctly.

ulferts commented:
HDinger#6 should fix the table width problems.

NobodysNightmare commented on app/views/work_packages/reports/report.html.erb:
```diff
@@ -28,21 +28,10 @@ See doc/COPYRIGHT.rdoc for more details.
-  </div>
-</div>
+<%  reports = [@author_report, @type_report, @priority_report, @version_report, @category_report, @assignee_report, @responsible_report]
+    reports <<  @subproject_report if @project.children.any? %>
```
I know, you just moved that code around. But wouldn't it be better, if the `ReportsController` would provide the list of reports (`@reports`?) directly, instead of providing several reports that the view has to know by name?

NobodysNightmare replied:
I've added a proposal as PR against your PR: https://github.com/HDinger/openproject/pull/7

NobodysNightmare committed:
move assembly of reports list into controller

NobodysNightmare commented:
LGTM :+1: 

Not sure if travis has to tell anything, but lets wait for him
